applications/HDF5Application/custom_io/hdf5_properties_io.cpp

```diff
@@ -49,20 +72,21 @@ void WriteProperties(File& rFile, std::string const& rPrefix, PropertiesType con
     KRATOS_CATCH("");
 }
 
-void WriteProperties(File& rFile, std::string const& rPrefix, PropertiesContainerType const& rProperties)
+void WriteProperties(
+    File& rFile,
+    const std::string& rPrefix,
+    const PropertiesContainerType & rProperties)
 {
     KRATOS_TRY;
 
     Vector<int> prop_ids(rProperties.size());
     unsigned i = 0;
-    for (const PropertiesType& r_properties : rProperties)
-    {
+    for (const PropertiesType& r_properties : rProperties) {
         prop_ids[i++] = r_properties.Id();
         WriteProperties(rFile, rPrefix, r_properties);
     }
     rFile.AddPath(rPrefix + "/Properties");
-    if (rProperties.size() > 0) // H5Awrite fails for empty container (sub model parts)
-    {
+    if (rProperties.size() > 0) { // H5Awrite fails for empty container (sub model parts)
         rFile.WriteAttribute(rPrefix + "/Properties", "Ids", prop_ids);
     }
```

Comment:
smells like a potential source of hanging. Who invokes `WriteProperties`? Is it ever possible that some ranks have properties to write while others don't?

Reply:
fixed with serialization now.